File: python/Problem012.py
#76576500 in 11.66162109375 seconds
import time

from EulerUtils import *

start = time.time()
count = 0
for i in range(1, 99999999, 1):
    count += i
    if is_prime(count) == False:
        if len(get_factors(count)) > 500:
            end = time.time()
            print(count,"in",end-start,"seconds")
            break

File: java/Problem012.java
package main;

public class Problem012 {

	public static void main(String args[]) {
		System.out.println(new Problem012().run());
	}
	
	public int run() {
		int count = 0;
		for(int i = 0; i < 99999999; i++) {
			count += i;
			if(!Utils.isPrime(count)) {
				if(Utils.getNumFactors(count) > 500) {
					return count;
				}
			}
		}
		return 0;
	}
}
